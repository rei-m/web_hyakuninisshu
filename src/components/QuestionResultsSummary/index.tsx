import * as React from 'react';
import styled from 'styled-components';
import { withAppTheme } from '@src/styles';

export interface QuestionResultsSummaryProps {
  readonly title: string;
  readonly value: string;
  readonly style?: React.CSSProperties;
}

const Root = withAppTheme(styled.div)`
  background-color: ${({ theme }) => theme.colorThin};
  box-sizing: border-box;
  padding: ${({ theme }) => theme.spacing1x};
  box-shadow: ${({ theme }) => theme.elevationShadow1x};
  border-radius: 4px;
`;

const Title = styled.div`
  text-align: left;
  font-size: 1.2rem;
`;

const Value = styled.div`
  font-size: 2.8rem;
`;

const QuestionResultsSummary: React.SFC<QuestionResultsSummaryProps> = ({
  title,
  value,
  style
}) => (
  <Root style={style}>
    <Title>{title}</Title>
    <Value>{value}</Value>
  </Root>
);

export default QuestionResultsSummary;
