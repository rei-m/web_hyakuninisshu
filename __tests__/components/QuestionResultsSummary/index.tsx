import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import QuestionResultsSummary from '@src/components/QuestionResultsSummary';

describe('<QuestionResultsSummary />', () => {
  it('should render component', () => {
    const renderer = ReactTestRenderer.create(
      <QuestionResultsSummary title="平均解答時間" value="5秒" />
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
