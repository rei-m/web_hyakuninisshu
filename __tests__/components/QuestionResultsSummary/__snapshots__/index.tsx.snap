// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionResultsSummary /> should render component 1`] = `
<div
  className="sc-bdVaJa jXieMO"
  style={undefined}
>
  <div
    className="sc-bwzfXH ijMQBI"
  >
    平均解答時間
  </div>
  <div
    className="sc-htpNat hHyKqi"
  >
    5秒
  </div>
</div>
`;
